### ISSUE TITLE
Stdout and stderr from sbt commands is gone forever

### ISSUE THREAD

**Issue by BritishPearl** [cb372/sbt-client#11]
The underlying `sbt` server is launched with stdin/out/err all pointing to /dev/null:

```rust
    Command::new("sbt")
        .stdin(Stdio::null())
        .stdout(Stdio::null())
        .stderr(Stdio::null())
        .spawn()
```

and thus the output from commands like `sbt test` or `sbt it` is unavailable, only the logs (`[info]` etc.). Is there any way to circumvent this limitation?

**Comment by cb372** [comment]
Thanks for the report. How about adding command-line options to write stdout and stderr to files?

e.g. `sbt-client -stdout out.txt -stderr err.txt test`

The only problem is that these options would be ignored if the sbt server is already running, which is potentially confusing behaviour.

**Comment by cb372** [comment]
See also #1 and #7, which are variants of the same problem: any options we take on the command line will only take effect if sbt-client starts an sbt server because it's not already running. Unless we can somehow detect if an sbt server is running, but with different arguments? In that case we could kill the server and start a new one.

**Comment by dwijnand** [comment]
You might want to hold off introducing new UX here, given https://github.com/sbt/sbt/pull/5319 is coming (at some point, hopefully), which would mostly subsume this client (AFAICT).

**Comment by BritishPearl** [comment]
@cb372 yes, `-stdout`/`-stdin` options would be nice... or even just `-inherit-stdout`/`-inherit-stdin` that would make sbt-client use `Stdio::inherit` instead of `Stdio::null` :)

**Comment by BritishPearl** [comment]
If it comes to server already running... yeah, possibly a check of the arguments of the existing server instance + an option like `-restart-if-different-options`
